api/v1alpha2/gcpmachine_types.go

```diff
@@ -84,7 +83,7 @@ type GCPMachineSpec struct {
 	// ServiceAccounts specify the service account email and which scopes to assign to the machine.
 	// Defaults to: email: "default", scope: []{compute.CloudPlatformScope}
 	// +optional
-	ServiceAccounts []*compute.ServiceAccount `json:"serviceAccounts,omitempty"`
+	ServiceAccounts []ServiceAccount `json:"serviceAccounts,omitempty"`
```

Comment:
There can be only a single ServiceAccount associated to a Machine, we should switch this to a pointer, wdyt?